src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy

```diff
@@ -104,12 +130,26 @@ class IqPolicyEvaluatorWorkflowStep
     }
 
     @Override
-    FormValidation doCheckIqApplication(@QueryParameter String value) {
+    FormValidation doCheckListAppId(@QueryParameter String value) {
       FormValidation.validateRequired(value)
     }
 
     @Override
-    ListBoxModel doFillIqApplicationItems(@QueryParameter @Nullable String jobCredentialsId, @AncestorInPath Job job) {
+    FormValidation doCheckManualAppId(@QueryParameter String value, @QueryParameter String jobCredentialsId,
+                                      @AncestorInPath Job job)
+    {
+      FormValidation val = FormValidation.validateRequired(value)
+      if (FormValidation.ok() == val) {
+        if (!IqUtil.
+            verifyOrCreateApplication(NxiqConfiguration.serverUrl.toString(), jobCredentialsId, job, value)) {
+          FormValidation.error(Messages._IqPolicyEvaluation_ManualApplicationVerificationFailed())
```

Comment:
I believe your intent is to _return_ the result of `FormValidation.error()` here rather than call it and discard the return value? If so, you need to either do an explicit `return` or assign this to `val` so that it will be returned below.